app/src/main/java/de/westnordost/streetcomplete/data/osm/osmquests/OsmQuestController.kt

```diff
@@ -222,6 +229,18 @@ import javax.inject.Singleton
         val countries = questType.enabledInCountries
         if (!countryBoundariesFuture.get().isInAny(pos, countries))  return false
 
+        // do not create quests outside of hours specified
+        val dayNightCycle = questType.dayNightCycle
+
+        if (!prefs.getBoolean(Prefs.ALWAYS_SHOW_TIME_DEPENDENT, false)) {
+            if (!when (dayNightCycle) {
+                    is DayAndNight -> true
+                    is OnlyDay -> !isNight(pos)
+                    is OnlyNight -> isNight(pos)
+                }
+            ) return false
+        }
+
```

Comment:
The logic should go into an own class. For example `DayNightQuestFilter`. This class could have just one method: `isVisible(quest: Quest): Boolean`

Also, the the `dayNightCycle` is not necessarily limited to OSM quests, so the better place for this would be `VisibleQuestsSource`

Reply:
How can I query OsmElementQuestType variables like the new `dayNightCycle` I declared in [OsmElementQuestType.kt#L29](https://github.com/streetcomplete/StreetComplete/blob/ecb25f986ed93ca8ad4d635a0dd83bda9c7d7185/app/src/main/java/de/westnordost/streetcomplete/data/osm/osmquests/OsmElementQuestType.kt#L29) from the Quest object given to isVisible? 

I have tried `quest.type.dayNightCycle` but Android Studio is saying no 😭

Reply:
Because you added the property to `OsmElementQuestType.kt`, not `QuestType.kt`

Reply:
If it's added to QuestType.kt will it still work the same? Are there any quests that aren't OsmElementQuestType's? :)

Reply:
currently, only note quests